src/evict/evict_lru.c

```diff
@@ -354,6 +354,33 @@ __evict_thread_run(WT_SESSION_IMPL *session, WT_THREAD *thread)
     return (ret);
 }
 
+/*
+ * __evict_set_saved_walk_tree --
+ *     Set saved walk tree maintaining use count. Call it with NULL to clear the saved walk tree.
+ */
+static void
+__evict_set_saved_walk_tree(WT_SESSION_IMPL *session, WT_DATA_HANDLE *new_dhandle)
+{
+    WT_CACHE *cache;
+    WT_DATA_HANDLE *walk_tree;
+
+    cache = S2C(session)->cache;
+    walk_tree = cache->walk_tree;
+
+    if (walk_tree == new_dhandle)
+        return;
+
+    if (walk_tree != NULL) {
+        WT_ASSERT(session, __wt_atomic_loadi32(&walk_tree->session_inuse) > 0);
+        (void)__wt_atomic_subi32(&walk_tree->session_inuse, 1);
+    }
+
+    cache->walk_tree = new_dhandle;
+
+    if (new_dhandle != NULL)
```

Comment:
nit: can we increase this number before this line:

```
cache->walk_tree = new_dhandle;
```

Reply:
Done